cli/Main.hs

```diff
@@ -49,9 +59,16 @@ main' inplace opts mfilepath input = do
     res <- runCabalFmtIO (takeDirectory <$> mfilepath) opts (cabalFmt filepath input)
 
     case res of
-        Right output
-            | inplace   -> writeFile filepath output
-            | otherwise -> putStr output
+        Right output -> do
+            let outputBS = toUTF8BS output
+                formatted = outputBS == input
+
+            if inplace
+            then unless formatted $ BS.writeFile filepath outputBS
+            else BS.putStr outputBS
```

Comment:
This seems wrong. If the inplace flag is on, the `formatted` doesn't have any affect?

Reply:
If the filer is already formatted, then no need to overwrite it, no?